src/workerd/api/http.c++

```diff
@@ -1330,8 +1352,13 @@ void Request::serialize(jsg::Lock& js,
             //
             // Note we have to double-Maybe this, so that if no signal is present, the property is absent
             // instead of `null`.
-            .signal = signal.map([](jsg::Ref<AbortSignal>& s) -> kj::Maybe<jsg::Ref<AbortSignal>> {
-    return s.addRef();
+            .signal =
+                signal.map([&js](jsg::Ref<AbortSignal>& s) -> kj::Maybe<jsg::Ref<AbortSignal>> {
```

Comment:
This stuff gives me a headache. Maybe we should be serializing this signal instead of signal? Otherwise the incoming request's signal would never get passed over RPC in the first place.

Reply:
I don't really understand what you mean here. Can you elaborate?

Reply:
I pushed a commit to move the incoming request's signal to the `signal` field instead of `thisSignal` as I think it made the code simpler.